django/contrib/admin/static/admin/js/SelectBox.js

```diff
@@ -1,136 +1,126 @@
+                SelectBox.delete_from_cache(from, this.value);
+            });
             SelectBox.redisplay(from);
             SelectBox.redisplay(to);
         },
-        move_all: function(from, to) {
-            var from_box = document.getElementById(from);
-            var option;
-            var boxOptions = from_box.options;
-            var boxOptionsLength = boxOptions.length;
-            for (var i = 0, j = boxOptionsLength; i < j; i++) {
-                option = boxOptions[i];
-                var option_value = option.value;
-                if (SelectBox.cache_contains(from, option_value)) {
-                    SelectBox.add_to_cache(to, {value: option_value, text: option.text, displayed: 1});
-                    SelectBox.delete_from_cache(from, option_value);
-                }
-            }
+        move_all: function (from, to) {
+            $("#" + from + " option").each(function () {
+                var group = this.parentNode.tagName.toLowerCase() == 'optgroup' ? this.parentNode.getAttribute('label') : null;
+                SelectBox.add_to_cache(to, group, { value: this.value, text: this.text, displayed: 1 });
+                SelectBox.delete_from_cache(from, this.value);
+            });
             SelectBox.redisplay(from);
             SelectBox.redisplay(to);
         },
-        sort: function(id) {
-            SelectBox.cache[id].sort(function(a, b) {
-                a = a.text.toLowerCase();
-                b = b.text.toLowerCase();
-                if (a > b) {
-                    return 1;
-                }
-                if (a < b) {
-                    return -1;
+        sort: function (id, group) {
+            var cache = SelectBox.cache[id];
+            for (var i = 0; i < cache.length; i++) {
+                if (group === false || cache[i].group == group) {
+                    cache[i].items.sort(function (a, b) {
+                        a = a.text.toLowerCase();
+                        b = b.text.toLowerCase();
+                        try {
+                            if (a > b) {
+                                return 1;
+                            }
+                            if (a < b) {
+                                return -1;
+                            }
+                        } catch (e) {
+                            // silently fail on IE 'unknown' exception
+                        }
+                        return 0;
+                    });
                 }
-                return 0;
-            } );
+            }
         },
         select_all: function(id) {
-            var box = document.getElementById(id);
-            var boxOptions = box.options;
-            var boxOptionsLength = boxOptions.length;
-            for (var i = 0; i < boxOptionsLength; i++) {
-                boxOptions[i].selected = 'selected';
-            }
+            $("#" + id + " option").attr('selected', 'selected');
```

Comment:
This change is unrelated. It is restoring usage of jQuery which is undesirable.

Reply:
restored